libtenzir/builtins/operators/every_cron.cpp

```diff
@@ -218,6 +233,49 @@ class every_plugin final : public virtual operator_plugin<every_operator> {
   }
 
   auto parse_operator(parser_interface& p) const -> operator_ptr override {
+    using operator_type = scheduled_execution_operator<Scheduler>;
+    auto scheduler = Scheduler::parse(p);
+
+    auto result = p.parse_operator();
+    if (not result.inner) {
+      diagnostic::error("failed to parse operator")
+        .primary(result.source)
+        .throw_();
+    }
+    if (auto* pipe = dynamic_cast<pipeline*>(result.inner.get())) {
+      auto ops = std::move(*pipe).unwrap();
+      for (auto& op : ops) {
+        op = std::make_unique<operator_type>(std::move(op),
+                                             std::move(scheduler));
+        // Only the first operator can be a source and needs to be replaced.
+        break;
+      }
+      return std::make_unique<pipeline>(std::move(ops));
+    }
+    return std::make_unique<operator_type>(std::move(result.inner),
+                                           std::move(scheduler));
+  }
+};
+
+/// scheduler for the 'every' operator
+class every_scheduler {
+public:
+  every_scheduler() = default;
+  explicit every_scheduler(const duration interval) : interval_{interval} {
+  }
+
+  constexpr static std::string_view name = "every";
+
+  auto field(auto& f) {
+    return f.field("interval", interval_);
+  }
+
+  auto next_after(time::clock::time_point now) const noexcept {
+    return std::chrono::time_point_cast<time::clock::time_point::duration>(
+      now + interval_);
+  }
```

Comment:
Please always add a trailing return type, unless it's not easily possibly. 

Also, while there was a time where we added `noexcept` everywhere, we stopped doing that because there is little gain. Also, it does not play well with `TENZIR_ASSERT` (which, yes, uses exceptions).